## Homework

### Set up the environment

You need to install Python, NumPy, Pandas, Matplotlib and Seaborn. For that, you can use the instructions from
[06-environment.md](../../../01-intro/06-environment.md).

### Q1. Pandas version

What's the version of Pandas that you installed?

You can get the version information using the `__version__` field:

```python
pd.__version__
```


In [2]:
import pandas as pd 

pd.__version__

'2.3.2'


### Getting the data 

For this homework, we'll use the Car Fuel Efficiency dataset. Download it from <a href='https://raw.githubusercontent.com/alexeygrigorev/datasets/master/car_fuel_efficiency.csv'>here</a>.

You can do it with wget:
```bash
wget https://raw.githubusercontent.com/alexeygrigorev/datasets/master/car_fuel_efficiency.csv

In [4]:
data = pd.read_csv('https://raw.githubusercontent.com/alexeygrigorev/datasets/master/car_fuel_efficiency.csv')
data.describe()

engine_displacement  num_cylinders   horsepower  vehicle_weight  \
count          9704.000000    9222.000000  8996.000000     9704.000000   
mean            199.708368       3.962481   149.657292     3001.280993   
std              49.455319       1.999323    29.879555      497.894860   
min              10.000000       0.000000    37.000000      952.681761   
25%             170.000000       3.000000   130.000000     2666.248985   
50%             200.000000       4.000000   149.000000     2993.226296   
75%             230.000000       5.000000   170.000000     3334.957039   
max             380.000000      13.000000   271.000000     4739.077089   

       acceleration   model_year    num_doors  fuel_efficiency_mpg  
count   8774.000000  9704.000000  9202.000000          9704.000000  
mean      15.021928  2011.484027    -0.006412            14.985243  
std        2.510339     6.659808     1.048162             2.556468  
min        6.000000  2000.000000    -4.000000             6.200971  
25%       13.300000  2006.000000    -1.000000            13.267459  
50%       15.000000  2012.000000     0.000000            15.006037  
75%       16.700000  2017.000000     1.000000            16.707965  
max       24.300000  2023.000000     4.000000            25.967222

### Q2. Records count

How many records are in the dataset?

- 4704
- 8704
- 9704
- 17704

In [5]:
data.shape[0]

9704

### Q3. Fuel types

How many fuel types are presented in the dataset? 

- 1
- 2
- 3
- 4

In [9]:
data.columns
data['fuel_type'].unique()


array(['Gasoline', 'Diesel'], dtype=object)

In [11]:
data.columns

Index(['engine_displacement', 'num_cylinders', 'horsepower', 'vehicle_weight',
       'acceleration', 'model_year', 'origin', 'fuel_type', 'drivetrain',
       'num_doors', 'fuel_efficiency_mpg'],
      dtype='object')

### Q4. Missing values

How many columns in the dataset have missing values?

- 0
- 1
- 2
- 3
- 4

In [10]:
data.isnull().any(axis=1).sum()

np.int64(2371)

### Q5. Max fuel efficiency

What's the maximum fuel efficiency of cars from Asia?

- 13.75
- 23.75
- 33.75
- 43.75

In [1]:
data['origin'].unique()

NameError: name 'data' is not defined